modules/data-streams/src/internalClusterTest/java/org/elasticsearch/datastreams/FailureStoreMetricsWithIncrementalBulkIT.java

```diff
@@ -0,0 +1,283 @@
+        assertThat(getDataStreamResponse.getDataStreams().get(0).getDataStream().getIndices().size(), equalTo(1));
+        String backingIndex = getDataStreamResponse.getDataStreams().get(0).getDataStream().getIndices().get(0).getName();
+        assertThat(backingIndex, backingIndexEqualTo(dataStream, 1));
+
+        String node = findShard(resolveIndex(backingIndex), 0);
+        IndexingPressure primaryPressure = internalCluster().getInstance(IndexingPressure.class, node);
+        long memoryLimit = primaryPressure.stats().getMemoryLimit();
+        long primaryRejections = primaryPressure.stats().getPrimaryRejections();
+        try (Releasable releasable = primaryPressure.markPrimaryOperationStarted(10, memoryLimit, false)) {
+            while (primaryPressure.stats().getPrimaryRejections() == primaryRejections) {
+                while (nextRequested.get()) {
+                    nextRequested.set(false);
+                    refCounted.incRef();
+                    List<DocWriteRequest<?>> requests = new ArrayList<>();
+                    for (int i = 0; i < 20; ++i) {
+                        requests.add(indexRequest(dataStream));
+                    }
+                    handler.addItems(requests, refCounted::decRef, () -> nextRequested.set(true));
+                }
+                assertBusy(() -> assertTrue(nextRequested.get()));
+            }
+        }
+
+        while (nextRequested.get()) {
+            nextRequested.set(false);
+            refCounted.incRef();
+            handler.addItems(List.of(indexRequest(dataStream)), refCounted::decRef, () -> nextRequested.set(true));
+        }
+
+        assertBusy(() -> assertTrue(nextRequested.get()));
+
+        PlainActionFuture<BulkResponse> future = new PlainActionFuture<>();
+        handler.lastItems(List.of(indexRequest(dataStream)), refCounted::decRef, future);
+
+        BulkResponse bulkResponse = safeGet(future);
+
+        /* The datastream should have a failure store associated with it now */
+        getDataStreamRequest = new GetDataStreamAction.Request(TEST_REQUEST_TIMEOUT, new String[] { "*" });
+        getDataStreamResponse = client().execute(GetDataStreamAction.INSTANCE, getDataStreamRequest).actionGet();
+        Index failureStoreIndex = getDataStreamResponse.getDataStreams().get(0).getDataStream().getFailureStoreWriteIndex();
+        String failureStoreIndexName = failureStoreIndex.getName();
+        String failure_node = findShard(resolveIndex(failureStoreIndexName), 0);
+        boolean shardsOnDifferentNodes = node.equals(failure_node) == false;
+
+        for (int i = 0; i < hits.get(); ++i) {
+            assertFalse(bulkResponse.getItems()[i].isFailed());
+            assertTrue(bulkResponse.getItems()[i].getFailureStoreStatus().getLabel().equalsIgnoreCase("NOT_APPLICABLE_OR_UNKNOWN"));
+        }
+
+        int docs_redirected_to_fs = 0;
+        int docs_in_fs = 0;
```

Comment:
Java variables should be camel case. `docsInFS`

Reply:
Done.